src/test/java/org/folio/rest/api/HoldingsSourceTest.java

```diff
@@ -0,0 +1,308 @@
+package org.folio.rest.api;
+
+import static org.folio.rest.support.http.InterfaceUrls.holdingsSourceUrl;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.folio.rest.support.IndividualResource;
+import org.folio.rest.support.Response;
+import org.folio.rest.support.ResponseHandler;
+import org.folio.rest.support.builders.HoldingRequestBuilder;
+import org.folio.rest.support.http.ResourceClient;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+
+public class HoldingsSourceTest extends TestBaseWithInventoryUtil {
+  private static ResourceClient holdingsSourceClient;
+
+  @BeforeClass
+  public static void beforeAll() {
+    holdingsSourceClient = ResourceClient.forHoldingsSource(client);
+  }
+
+  @Test
+  public void canCreateHoldingsSource()
+    throws MalformedURLException, InterruptedException,
+    ExecutionException, TimeoutException {
+
+    UUID sourceId = UUID.randomUUID();
+
+    JsonObject source = holdingsSourceClient.create(
+      new JsonObject()
+      .put("id", sourceId.toString())
+      .put("name", "test source")
```

Comment:
I'd be tempted to introduce a builder to reduce the low level detail of the structure in every test.

I'd be more than happy if that was using lombok to reduce the need for much code.

Reply:
Does this need to happen in order to merge the PR?

Reply:
No. I think it would simplify the tests and make them more maintainable. It is only a suggestion, and if you want to ignore it or defer it, that is ok.